src/unittest/test.h

```diff
@@ -277,3 +277,45 @@ class message_collector : public dpp::message_collector {
 		set_test("MSGCOLLECT", list.size() > 0);
 	}
 };
+
+/**
+ * @brief Convenience functor to get the snowflake of a certain type
+ */
+struct user_project_id_t {
+	dpp::snowflake operator()(const dpp::user &user) const noexcept {
+		return user.id;
+	}
+
+	dpp::snowflake operator()(const dpp::guild_member &user) const noexcept {
+		return user.user_id;
+	}
+
+	dpp::snowflake operator()(dpp::snowflake user) const noexcept {
+		return user;
+	}
+
+	dpp::snowflake operator()(const dpp::thread_member &user) const noexcept {
+		return user.user_id;
+	}
+};
+
+/**
+ * @brief Convenience lambda to get the user snowflake of a certain user type
+ * @see user_project_id_t
+ */
+inline constexpr user_project_id_t get_user_snowflake;
+
+/**
+ * @brief Convenience lambda to check if a certain user is the owner of the test bot, mostly meant to be passed to standard algorithms
+ * @see get_user_snowflake
+ *
+ * @return bool whether the user is the test bot owner
+ */
+inline constexpr auto is_owner = [](auto &&user) noexcept -> bool {
```

Comment:
please don't do this! (return type auto with actual return type on the end of the definition using ->) it may be modern but it is unreadable to me, and doesn't follow the code style of the rest of dpp.

Reply:
(besides it's just extra typing and forces visual scanning all the way across the line, what's wrong with `constexpr bool is_owner...`

Reply:
This is lambda, and it needs to be a lambda to be passed for example to std::find_if